src/Momento.Sdk/Internal/AuthGrpcManager.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Net.Client;
+#if USE_GRPC_WEB
+using System.Net.Http;
+using Grpc.Net.Client.Web;
+#endif
+using Microsoft.Extensions.Logging;
+using Momento.Protos.TokenClient;
+using Momento.Sdk.Config;
+using Momento.Sdk.Config.Middleware;
+using Momento.Sdk.Internal.Middleware;
+using static System.Reflection.Assembly;
+
+namespace Momento.Sdk.Internal;
+
+public interface IAuthClient
+{
+    public Task<_GenerateDisposableTokenResponse> generateDisposableToken(_GenerateDisposableTokenRequest request, CallOptions callOptions);
+}
+
+internal class AuthClientWithMiddleware : IAuthClient
+{
+    private readonly IList<IMiddleware> _middlewares;
+    private readonly Token.TokenClient _generatedClient;
+    private readonly IList<Tuple<string, string>> _headers;
+
+    public AuthClientWithMiddleware(Token.TokenClient generatedClient, IList<IMiddleware> middlewares, IList<Tuple<string, string>> headers)
+    {
+        _generatedClient = generatedClient;
+        _middlewares = middlewares;
+        _headers = headers;
+    }
+
+    public async Task<_GenerateDisposableTokenResponse> generateDisposableToken(_GenerateDisposableTokenRequest request, CallOptions callOptions)
+    {
+        var wrapped = await _middlewares.WrapRequest(request, callOptions, (r, o) => _generatedClient.GenerateDisposableTokenAsync(r, o));
+        return await wrapped.ResponseAsync;
+    }
+
+}
+
+public class AuthGrpcManager : IDisposable
+{
+    private readonly GrpcChannel channel;
+    public IAuthClient Client { get; }
+
+#if USE_GRPC_WEB
```

Comment:
is having a web endpoint necessary for dotnet? 
I thought it was in the JS SDK because of the split between web vs nodejs packages

Reply:
we actually do support grpc-web in .NET, because some really old versions of the .NET framework (and possibly also Unity) do not support HTTP 2.0.